internal/pkg/dashboard/dashboard.go

```diff
@@ -46,14 +48,24 @@ const (
 	ScreenNetworkConfig Screen = "Network Config"
 )
 
-// DataWidget is a widget which consumes Data to draw itself.
-type DataWidget interface {
-	Update(node string, data *data.Data)
+// APIDataListener is a listener which is notified when API-sourced data is updated.
+type APIDataListener interface {
+	OnAPIDataChange(node string, data *apidata.Data)
 }
 
-// LogWidget is a widget which consumes logs and draws itself.
-type LogWidget interface {
-	UpdateLog(node string, logLine string)
+// ResourceDataListener is a listener which is notified when a resource is updated.
+type ResourceDataListener interface {
+	OnResourceDataChange(data resourcedata.Data)
+}
+
+// LogDataListener is a listener which is notified when a log line is received.
+type LogDataListener interface {
+	OnLogDataChange(node string, logLine string)
+}
+
+// NodeSetListener is a listener which is notified when the set of nodes changes.
+type NodeSetListener interface {
+	OnNodeSetChange(nodes []string)
 }
```

Comment:
The idea here is to make the 3 different data sources and their listeners namings similar to each other.